example/lib/pages/circle.dart

```diff
@@ -1,49 +1,187 @@
 
+  @override
+  State<CirclePage> createState() => _CirclePageState();
+}
+
+class _CirclePageState extends State<CirclePage> {
+  final LayerHitNotifier<HitValue> _hitNotifier = ValueNotifier(null);
+  List<HitValue>? _prevHitValues;
+  List<CircleMarker<HitValue>>? _hoverCircles;
+
+  final _circlesRaw = <CircleMarker<HitValue>>[
+    CircleMarker(
+      point: const LatLng(51.5, -0.09),
+      color: Colors.white.withOpacity(0.7),
+      borderColor: Colors.black,
+      borderStrokeWidth: 2,
+      useRadiusInMeter: false,
+      radius: 100,
+      hitValue: (title: 'White', subtitle: 'Radius in logical pixels'),
+    ),
+    CircleMarker(
+      point: const LatLng(51.5, -0.09),
+      color: Colors.black.withOpacity(0.7),
+      borderColor: Colors.black,
+      borderStrokeWidth: 2,
+      useRadiusInMeter: false,
+      radius: 50,
+      hitValue: (
+        title: 'Black',
+        subtitle: 'Radius in logical pixels, should be above white.',
+      ),
+    ),
+    CircleMarker(
+      point: const LatLng(51.4937, -0.6638),
+      // Dorney Lake is ~2km long
+      color: Colors.green.withOpacity(0.9),
+      borderColor: Colors.black,
+      borderStrokeWidth: 2,
+      useRadiusInMeter: true,
+      radius: 1000, // 1000 meters
+      hitValue: (
+        title: 'Green',
+        subtitle: 'Radius in meters, calibrated over ~2km rowing lake'
+      ),
+    ),
+  ];
+  late final _circles =
+      Map.fromEntries(_circlesRaw.map((e) => MapEntry(e.hitValue, e)));
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(title: const Text('Circle')),
-      drawer: const MenuDrawer(route),
+      appBar: AppBar(title: const Text('Circles')),
+      drawer: const MenuDrawer(CirclePage.route),
       body: FlutterMap(
         options: const MapOptions(
           initialCenter: LatLng(51.5, -0.09),
           initialZoom: 11,
         ),
         children: [
           openStreetMapTileLayer,
-          CircleLayer(
-            circles: [
-              CircleMarker(
-                point: const LatLng(51.5, -0.09),
-                color: Colors.blue.withOpacity(0.7),
-                borderColor: Colors.black,
-                borderStrokeWidth: 2,
-                useRadiusInMeter: true,
-                radius: 2000, // 2000 meters
+          MouseRegion(
+            hitTestBehavior: HitTestBehavior.deferToChild,
+            cursor: SystemMouseCursors.click,
+            onHover: (_) {
+              final hitValues = _hitNotifier.value?.hitValues.toList();
+              if (hitValues == null) return;
+
+              if (listEquals(hitValues, _prevHitValues)) return;
+              _prevHitValues = hitValues;
+
+              final hoverCircles = hitValues.map((v) {
+                final original = _circles[v]!;
+
+                return CircleMarker<HitValue>(
+                  point: original.point,
+                  radius: original.radius + (15 / 2) - 1,
```

Comment:
Is this `(15 / 2) - 1` calculation only to set the hover circles bigger than the regular circles?

Reply:
Yep, just shifts the hover outline out (so its 15px out, not 7.5 px out & 7.5 px in). Just a visual thing, can remove if you want.

Reply:
```suggestion
                  radius: original.radius + 6.5,
```

Reply:
I would perhaps not suggest doing this. It will be optimized out by the compiler, there's no extra cost, and it kinda makes it a little bit more clear what's going on leaving it extended. But I don't really care that much :D

Reply:
True, but because this is our example app where people can view example implementations I prefer to have it's implementation clean. (: